### ISSUE TITLE
error in reshaping wannier.mlwfP into (nCells,nBands, nBands) in WannierImEps.py

### ISSUE THREAD

**Issue by goudarzi58** [shankar1729/jdftx#72]
Hi Shankar,

I'm trying to calculate the imaginary part of the dielectric function of gold using WannierImEps.py shared on the website but I'm getting the following error related to reshaping Pwannier.

#Pwannier = np.reshape(Pwannier, (nCells,nBands, nBands))---->
_**ValueError: cannot reshape array of size 259956 into shape (2407,6,6)_**

I'm not sure if the input files have resulted in wrong calculations. I was wondering if you could check out at input data and let me know about the error there.

Best,
Abbas

#######################################
The information of the generated files are:
cellMap: <class 'numpy.ndarray'> cellMap.shape: (2407, 3)
Hwannier: <class 'numpy.ndarray'> Hwannier.shape: (86652,)
Pwannier: <class 'numpy.ndarray'> Pwannier.shape: (259956,)
nCells: 2407
nBands: 6
#######################################
and the input files for JDFTx are:

#######    common.in

lattice face-centered Cubic 7.6817367
ion-species SG15/$ID_ONCV_PBE.upf
elec-cutoff 30
ion Au 0.00 0.00 0.00 0

#######   totalE.in
include common.in
kpoint-folding 12 12 12
elec-n-bands 14
elec-smearing Fermi 0.01
electronic-SCF
converge-empty-states yes
dump-name totalE.$VAR
dump End State BandEigs ElecDensity
density-of-states Total
add-U Au d 0.07496862 #2.04eV
dump End EigStats

#########  wannier.in
include totalE.in
#inner and outer windows are important and have to be set correctly
wannier \
    innerWindow -0.042 0.780 \
    outerWindow -0.042 1.555 \
    saveMomenta yes

wannier-center Gaussian -0.101403  -0.435547   0.487558
wannier-center Gaussian -0.443339   0.055289  -0.446398
wannier-center Gaussian -0.361538  -0.010718   0.032397
wannier-center Gaussian -0.296948   0.326939  -0.400117
wannier-center Gaussian  0.030459  -0.158043  -0.194850
wannier-center Gaussian  0.030459   0.075321  -0.355555

wannier-initial-state totalE.$VAR
wannier-dump-name wannier.$VAR

wannier-minimize nIterations 300

**Comment by shankar1729** [comment]
Hi Abbas,

I'm slightly confused by the error above: the code that initially shows the error tries to reshape to (nCells, nBands, nBands), but the code in the tutorial asks you to reshape to (nCells,3,nBands,nBands), which is correct. Your correction in the second message basically restores the code to the version lredy on the tutorial webpage. Did I miss something there?

Also note that the newest version of the code changes the wannier output formats, and we have not yet updated these tutorials to match. However, you do not seem to be facing that issue yet.

Best,
Shankar

**Comment by goudarzi58** [comment]
I copied the code from the website, it's okay now. Actually, I need to work with JDFTx output data using python. It's a little bit hard to find out how the data are saved in JDFtx outputs and how it's possible to use them in python. I have been able to draw wannier band structures using Matplotlib. and still working to do the same with bandstructur.plot. I was wondering if you put some descriptions about each output and how they can be used in Python or Matlab while updating the website.

Best,
Abbas
